File: Solutions/0082- Remove Duplicates from Sorted List II/Solution.py
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        dummy = ListNode()
        dummy.next = head

        previous = dummy
        current = head

        while current:
            while current.next and current.next.val == current.val:
                current = current.next

            if previous.next == current:
                previous = current
            else:
                previous.next = current.next

            current = current.next

        return dummy.next

File: Solutions/0082- Remove Duplicates from Sorted List II/Solution.java
class Solution {
    public class ListNode {
        int val;
        ListNode next;
        ListNode() {}
        ListNode(int val) { this.val = val; }
        ListNode(int val, ListNode next) { this.val = val; this.next = next; }
    }
    
    public ListNode deleteDuplicates(ListNode head) {
        ListNode dummy = new ListNode(-200);
        var current = head;
        var pointer = dummy;

        while (current != null) {
            if (current.val != current.next.val) {
                pointer.next = current;
                current = current.next;
                pointer = pointer.next;
            } else {
                int number = current.val;
                while (current.val == number) {
                    current = current.next;
                }
            }

        }

        return dummy.next;
    }

}